src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -42,9 +44,22 @@ public CommandResult execute(Model model) throws CommandException {
 
         Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
         model.deletePerson(personToDelete);
+        removePersonFromEvent(model, personToDelete);
         return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, Messages.format(personToDelete)));
     }
 
+    /**
+     * Removes a specified person from all events they are attending in the model's event list.
+     */
+    private void removePersonFromEvent(Model model, Person personToRemove) {
+        ObservableList<Event> eventsList = model.getEventList();
+        for (Event event : eventsList) {
```

Comment:
We can simply remove with the ``if`` condition check, which simplifies the code and makes it more clear. ``Set.remove`` only modifies the set if element (person) is present in the set.